libsolidity/parsing/Parser.cpp

```diff
@@ -2204,6 +2206,23 @@ ASTPointer<ASTString> Parser::expectIdentifierToken()
 	return getLiteralAndAdvance();
 }
 
+ASTPointer<ASTString> Parser::expectIdentifierTokenOrAddress()
+{
+	ASTPointer<ASTString> result;
+	if (m_scanner->currentToken() == Token::Address)
+	{
+		result = make_shared<ASTString>("address");
+		m_scanner->next();
+	}
+	else
+	{
+		// do not advance on success
```

Comment:
This comment is confusing, given `getLiteralAndAdvance` does it already? I think the comment is warning that no `next()` is required after `getLiteralAndAdvance`, but I would just get rid of the comment.

Reply:
The comment explains the false flag. I will move it.